src/main.cpp

```diff
@@ -0,0 +1,235 @@
+    const boost::filesystem::path& keyFile)
+{
+    if (args.empty())
+        throw std::runtime_error ("no command specified");
+
+    using namespace ripple;
+
+    if (args.size() != 1)
+        throw std::runtime_error (
+            "Syntax effor: Wrong number of parameters");
+    
+    if (args[0] == "create_master_keys")
+        createKeyFile (keyFile);
+    else if (args[0] == "create_signing_keys")
+        signManifest (keyFile, boost::none /* sequence */);
+    else if (args[0] == "revoke_master_keys")
+        signManifest (keyFile, std::numeric_limits<std::uint32_t>::max ());
+    else
+        throw std::runtime_error ("Unknown command");
+
+    return 0;
+}
+
+static
+std::string
+getEnvVar (char const* name)
+{
+    std::string value;
+
+    auto const v = getenv (name);
+
+    if (v != nullptr)
+        value = v;
+
+    return value;
+}
+
+int main (int argc, char** argv)
+{
+#if defined(__GNUC__) && !defined(__clang__)
+    auto constexpr gccver = (__GNUC__ * 100 * 100) +
+                            (__GNUC_MINOR__ * 100) +
+                            __GNUC_PATCHLEVEL__;
+
+    static_assert (gccver >= 50100,
+        "GCC version 5.1.0 or later is required to compile rippled.");
+#endif
+
+    static_assert (BOOST_VERSION >= 105700,
+        "Boost version 1.57 or later is required to compile rippled");
+
+    po::variables_map vm;
+
+    // Set up option parsing.
+    //
+    po::options_description desc ("General Options");
+    desc.add_options ()
+    ("help,h", "Display this message.")
+    ("keyfile", po::value<std::string> (), "Specify the master key file.")
+    ("unittest,u", po::value <std::string> ()->implicit_value (""),
+        "Perform unit tests.")
+    ("parameters", po::value< std::vector<std::string> > (),
+        "Specify comma separated parameters.")
+    ;
+
+    // Interpret positional arguments as --parameters.
+    po::positional_options_description p;
+    p.add ("parameters", -1);
+
+    // Parse options, if no error.
+    try
+    {
+        po::store (po::command_line_parser (argc, argv)
+            .options (desc)               // Parse options.
+            .positional (p)               // Remainder as --parameters.
+            .run (),
+            vm);
+        po::notify (vm);                  // Invoke option notify functions.
+    }
+    catch (std::exception const&)
+    {
+        std::cerr << "manifest_tool: Incorrect command line syntax." << std::endl;
+        std::cerr << "Use '--help' for a list of options." << std::endl;
+        return 1;
+    }
+
+    // Run the unit tests if requested.
+    // The unit tests will exit the application with an appropriate return code.
+    if (vm.count ("unittest"))
+        return runUnitTests();
+
+    if (vm.count ("help") || ! vm.count ("parameters"))
+    {
+        printHelp (desc);
+        return 0;
+    }
+
+    const std::string DEFAULT_KEY_FILE =
+        getEnvVar ("HOME") + "/.ripple/validator-keys.json";
```

Comment:
I actually don't have a strong opinion about good behavior in this error condition.  I guess my first instinct was to error out if we don't get a good value from `"HOME"`.  But using the current directory would also be viable.  Mostly I just wanted to make sure we'd thought through what might happen and have a justifiable reaction.